src/_P123_SI7013.ino

```diff
@@ -0,0 +1,165 @@
+#include "_Plugin_Helper.h"
+
+#ifdef USES_P123
+#include "src/PluginStructs/P123_data_struct.h"
+//#######################################################################################################
+//######################## Plugin 123 SI7013 I2C Temperature Humidity Sensor  ###########################
+//#######################################################################################################
+// 08-06-2019 Florin Muntean
+
+#define PLUGIN_123
+#define PLUGIN_ID_123        123
+#define PLUGIN_NAME_123       "Environment - SI7013"
+#define PLUGIN_VALUENAME1_123 "Temperature"
+#define PLUGIN_VALUENAME2_123 "Humidity"
+#define PLUGIN_VALUENAME3_123 "ADC"
+
+
+# define P123_RESOLUTION      PCONFIG(0)
+# define P123_I2CADDR         PCONFIG(1)
+# define P123_FILTER_POWER    PCONFIG(2)
+
+
+//convert from I2C address to index
+uint8_t Plugin_123_device_index(const uint8_t i2caddr) {
+  switch(i2caddr) {
+    case SI7021_I2C_ADDRESS:  return 0u;
+    case SI7013_I2C_ADDRESS:  return 1u;
+  }
+  return 1u; // Some default
+}
+
+const uint8_t Plugin_123_i2c_addresses[2] = { SI7021_I2C_ADDRESS, SI7013_I2C_ADDRESS };
+
+boolean Plugin_123(byte function, struct EventStruct *event, String& string)
+{
+  boolean success = false;
+
+  switch (function)
+  {
+    case PLUGIN_DEVICE_ADD:
+      {
+        Device[++deviceCount].Number = PLUGIN_ID_123;
+        Device[deviceCount].Type = DEVICE_TYPE_I2C;
+        Device[deviceCount].VType = Sensor_VType::SENSOR_TYPE_TEMP_HUM_BARO; //using this type in order to send the 3rd value which is the ADC 
+        Device[deviceCount].Ports = 0;
+        Device[deviceCount].PullUpOption = false;
+        Device[deviceCount].InverseLogicOption = false;
+        Device[deviceCount].FormulaOption = true;
+        Device[deviceCount].ValueCount = 3;
+        Device[deviceCount].SendDataOption = true;
+        Device[deviceCount].TimerOption = true;
+        Device[deviceCount].GlobalSyncOption = true;
+        break;
+      }
+
+    case PLUGIN_GET_DEVICENAME:
+      {
+        string = F(PLUGIN_NAME_123);
+        break;
+      }
+
+    case PLUGIN_GET_DEVICEVALUENAMES:
+      {
+        strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_123));
+        strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[1], PSTR(PLUGIN_VALUENAME2_123));
+        strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[2], PSTR(PLUGIN_VALUENAME3_123));
+        break;
+      }
+
+    case PLUGIN_WEBFORM_LOAD:
+      {
+        #define SI7013_RESOLUTION_OPTIONS 4
+
+        byte choice = P123_RESOLUTION;
+        String options[SI7013_RESOLUTION_OPTIONS];
```

Comment:
Please use this instead for lists of options with only flash strings:
```c++
const __FlashStringHelper * options[SI7013_RESOLUTION_OPTIONS];
```
This does save a lot of memory on the ESP

Reply:
done